flocker/node/agents/ebs.py

```diff
@@ -107,6 +117,44 @@ def _wait_for_volume(expected_volume,
             )
 
 
+def _wait_for_new_device(base, time_limit=60):
+    """
+    Helper function to wait for up to 60s for new
+    EBS block device (`/dev/sd*` or `/dev/xvd*`) to
+    manifest in the OS.
+
+    :param list base: List of baseline block devices
+        that existed before execution of operation that expects
+        to create a new block device.
+
+    :returns unicode device: Name of the new block device
+        we are interested in.
+
+    :raises Exception: When new EBS block device did not
+        manifest in OS within time limit.
+
+    """
+    patterns = ['sd.*', 'xvd.*']
```

Comment:
These could be prefixes instead of patterns and then you could use `device.startswith(...)` instead of using regexp matching.  Less powerful - but still sufficiently powerful.  And it's nice to avoid regexps whenever possible.